angular/projects/global/src/lib/services/store.service.ts

```diff
@@ -0,0 +1,153 @@
+export class StoreService {
+  private currentState: any;
+
+  constructor() {
+    this.currentState = {};
+    this.state.navigationStack = [];
+  }
+
+  get state(): any {
+    return this.currentState;
+  }
+  set state(state: any) {
+    this[state.type] = state.data;
+  }
+
+  public getNavigationStack() {
+    if (this.state.navigationStack) {
+      return this.state.navigationStack;
+    } else {
+      return null;
+    }
```

Comment:
Do we need to check if `this.state` is non-null before checking `this.state.navigationStack`? Unless `this.state` can never be null.

(Optional) you could write it like this, as an alternate pattern to if/else.  But its the same output in the end in either case:
`return this.state && this.state.navigationStack || null`